packages/electron-updater/src/NsisUpdater.ts

```diff
@@ -27,6 +27,11 @@ export class NsisUpdater extends BaseUpdater {
       downloadUpdateOptions,
       fileInfo,
       task: async (destinationFile, downloadOptions, packageFile, removeTempDirIfAny) => {
+        if (isInvalid(destinationFile) || (packageFile != null && isInvalid(packageFile))) {
+          throw newError(`destinationFile or packageFile contains illegal chars`, "ERR_UPDATER_ILLEGAL_FILE_NAME")
+        }
```

Comment:
I believe we no longer to check for this invalid path, I've since opened a PR to escape these characters. See: https://github.com/electron-userland/electron-builder/pull/5116

Basically, throwing here means that we prevent user with username `.../User's/...` (_which is pretty common naming convention_) from updating and escaping would make more sense.